api/v1alpha1/clusterextension_types.go

```diff
@@ -83,6 +83,11 @@ type ClusterExtensionSpec struct {
 	//+kubebuilder:Optional
 	// Preflight defines the configuration of preflight checks.
 	Preflight *PreflightConfig `json:"preflight,omitempty"`
+
+	//+kubebuilder:validation:MaxLength:=48
```

Comment:
Service accounts can be any valid DNS 1123 subdomain. See https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names

We should adjust the MaxLength and Pattern to make sure it matches what DNS 1123 specifies.